packages/flutter_tools/test/integration.shard/migrate_status_test.dart

```diff
@@ -0,0 +1,202 @@
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+// import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/migrate.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../src/common.dart';
+import '../src/context.dart';
+import '../src/test_flutter_command_runner.dart';
+
+void main() {
+  FileSystem fileSystem;
+  BufferLogger logger;
+  Platform platform;
+  // TODO(garyq): Add terminal back in when other subcommands land.
+  // Terminal terminal;
+  ProcessManager processManager;
+  Directory appDir;
+
+  setUp(() {
+    fileSystem = globals.localFileSystem;
+    appDir = fileSystem.systemTempDirectory.createTempSync('apptestdir');
+    logger = BufferLogger.test();
+    platform = FakePlatform();
+    // terminal = Terminal.test();
+    processManager = globals.processManager;
+  });
+
+  setUpAll(() {
+    Cache.disableLocking();
+  });
+
+  tearDown(() async {
+    tryToDelete(appDir);
+  });
+
+  testUsingContext('Status produces all outputs', () async {
+    final MigrateCommand command = MigrateCommand(
+      verbose: true,
+      logger: logger,
+      fileSystem: fileSystem,
+      // terminal: terminal,
+      platform: platform,
+      processManager: processManager,
+    );
+    final Directory workingDir = appDir.childDirectory('migrate_working_dir');
+    appDir.childFile('lib/main.dart').createSync(recursive: true);
+    final File pubspecOriginal = appDir.childFile('pubspec.yaml');
+    pubspecOriginal.createSync();
+    pubspecOriginal.writeAsStringSync('''
+name: originalname
+description: A new Flutter project.
+version: 1.0.0+1
+environment:
+  sdk: '>=2.18.0-58.0.dev <3.0.0'
+dependencies:
+  flutter:
+    sdk: flutter
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+flutter:
+  uses-material-design: true''', flush: true);
+
+    final File pubspecModified = workingDir.childFile('pubspec.yaml');
+    pubspecModified.createSync(recursive: true);
+    pubspecModified.writeAsStringSync('''
+name: newname
+description: new description of the test project
+version: 1.0.0+1
+environment:
+  sdk: '>=2.18.0-58.0.dev <3.0.0'
+dependencies:
+  flutter:
+    sdk: flutter
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+flutter:
+  uses-material-design: false
+  EXTRALINE''', flush: true);
+
+    final File addedFile = workingDir.childFile('added.file');
+    addedFile.createSync(recursive: true);
+    addedFile.writeAsStringSync('new file contents');
+
+    final File manifestFile = workingDir.childFile('.migrate_manifest');
+    manifestFile.createSync(recursive: true);
+    manifestFile.writeAsStringSync('''
+merged_files:
+  - pubspec.yaml
+conflict_files:
+added_files:
+  - added.file
+deleted_files:
+''');
+
+    expect(appDir.childFile('lib/main.dart').existsSync(), true);
```

Comment:
is this needed? we explicitly created it on line 55, are we testing it didn't get deleted?